### ISSUE TITLE
Projection::invert fails with `Condition "0.0 == r3[3]" is true`

### ISSUE THREAD

**Issue by stuartcarnie** [godotengine/godot#100850]
### Tested versions

Godot Engine v4.4.dev7.official.46c8f8c5c - https://godotengine.org
Metal 3.2 - Forward+ - Using Device #0: Apple - Apple M1 Max (Apple7)

### System information

Godot v4.4.dev7 - macOS Sequoia (15.2.0) - Multi-window, 2 monitors - Metal (Forward+) - integrated Apple M1 Max (Apple7) - Apple M1 Max (10 threads)

### Issue description

Logs are spammed with the following:

```
ERROR: Condition "0.0 == r3[3]" is true.
   at: invert (/Volumes/Data/projects/games/godot/core/math/projection.cpp:758)
ERROR: Condition "0.0 == r3[3]" is true.
   at: invert (/Volumes/Data/projects/games/godot/core/math/projection.cpp:758)
```

### Steps to reproduce

1. Start TPS-Demo
2. Fire weapon so that it hits the ground[^1]
3. Shortly after bullet collides, logs will be spammed with the above errors

![Image](https://github.com/user-attachments/assets/344e5c9e-47e0-4e17-a2c9-9cf26978984f)

[^1]: It doesn't happen on all shots, and appears mostly when hitting the ground

### Minimal reproduction project (MRP)

TPS-Demo is where I observed this behaviour

**Comment by matheusmdx** [comment]
Oh my, someone that read the issue guidelines! Jokes aside, this seems to be a regression between the 4.4 dev 6 and 4.4 dev 7, i'll bisect right now.

**Comment by matheusmdx** [comment]
Bisected to #100319, @Flarkk

![Image](https://github.com/user-attachments/assets/994f0c20-d920-4a7d-949f-6c137d1a198c)

**Comment by Flarkk** [comment]
Just opened the fix #100862 for reviews. Should be okay now.
Yesterday night I was right on the above intuition, but wrong on the way to fix it.
